transfers/amclient.py

```diff
@@ -0,0 +1,518 @@
+        packages = packages + r['results']
+        if r['next']:
+            packages = self.get_all_packages(ptype, packages, r['next'])
+        return packages
+
+    def aip2dips(self):
+        """Get all DIPS created from AIP with UUID ``self.aip_uuid``."""
+        dips = self.get_all_dips()
+        return [d for d in dips if
+                d['current_path'].endswith(self.aip_uuid)]
+
+    def aips2dips(self):
+        """Get all AIP UUIDs and map them to their DIP UUIDs, if any."""
+        aips = self.get_all_aips()
+        dips = self.get_all_dips()
+        return dict([(a['uuid'],
+                      [d['uuid'] for d in dips if
+                       d['current_path'].endswith(a['uuid'])])
+                     for a in aips])
+
+    def download_dip(self):
+        """Download the DIP with UUID ``self.dip_uuid``."""
+        url = '{}/api/v2/file/{}/download/'.format(self.ss_url, self.dip_uuid)
+        r = requests.get(
+            url,
+            params={'username': self.ss_user_name,
+                    'api_key': self.ss_api_key},
+            stream=True)
+        print(r.status_code)
+        if r.status_code == 200:
+            local_filename = re.findall('filename="(.+)"',
+                                        r.headers['content-disposition'])[0]
+            with open(local_filename, 'wb') as f:
+                for chunk in r.iter_content(chunk_size=1024):
+                    if chunk:
+                        f.write(chunk)
+            print(local_filename)
+            return local_filename
+        else:
+            print('Unable to download DIP {}'.format(self.dip_uuid))
+
+    def reingest(self):
+        """Request that the Storage Service reingest the AIP with UUID
+        ``self.aip_uuid``. Example usage::
+
+            $ ./amclient.py reingest 198e7890-f19e-4b41-a172-953380e543e5 \
+                 --ss-api-key=fdbdf0f21196c6e1f9884452172e330ad41d8cae \
+                 --api-key=1b6aac9388733df195d4f63a8c6eb53dc1cffa31
+        """
+        url = '{}/api/v2/file/{}/reingest/'.format(self.ss_url, self.aip_uuid)
+        pipeline = self.aip_uuid2pipeline_uuid(self.aip_uuid)
+        # At present, the Storage Service will not make use of a
+        # 'processing_config' argument in a reingest_request.
+        payload = {
+            'pipeline': pipeline,
+            'reingest_type': 'objects',
+        }
+        auth_header = {'Authorization': 'ApiKey {}:{}'.format(
+                       self.ss_user_name, self.ss_api_key)}
+        r = requests.post(url, json=payload, headers=auth_header)
+        try:
+            assert r.status_code == 202
+            print('Reingest of AIP {} initiated.'.format(self.aip_uuid))
+        except AssertionError:
+            print('ERROR: something went wrong when attempting to reingest AIP'
+                  ' {}.'.format(self.aip_uuid))
+        self.approve_reingest(self.aip_uuid)
+
+    approve_reingest_retry_count = 5
```

Comment:
This constant should be inside the function that uses it, or at the top of the class with the rest of the constants. Since it's only used once I'm inclined to say the former.